pkg/ingester/ingester_v2.go

```diff
@@ -1981,28 +1987,47 @@ func (i *Ingester) v2LifecyclerFlush() {
 
 	ctx := context.Background()
 
-	i.compactBlocks(ctx, true)
+	i.compactBlocks(ctx, true, nil)
 	if i.cfg.BlocksStorageConfig.TSDB.IsBlocksShippingEnabled() {
-		i.shipBlocks(ctx)
+		i.shipBlocks(ctx, nil)
 	}
 
 	level.Info(i.logger).Log("msg", "finished flushing and shipping TSDB blocks")
 }
 
+const tenantParam = "t"
```

Comment:
I'm wondering why not using X-Scope-OrgID for specifying the tenant, like we did for the tenant deletion (even if it was an internal API). May be better for consistency.

If we keep the parameter, then I would call it explicitly `tenant` or `tenant_id` for better clarity.

Reply:
> I'm wondering why not using X-Scope-OrgID for specifying the tenant, like we did for the tenant deletion (even if it was an internal API). May be better for consistency.

There are two problems with using `X-Scope-OrgID` for this endpoint:
- we could only specify multiple tenants when using `-tenant-federation.enabled` flag, which is quite strange flag to enable on ingesters only to be supported by flush handler
- what is worse, by enabling authentication on flush endpoint, we would not be able to call this endpoint with no tenant to flush all users.

> If we keep the parameter, then I would call it explicitly `tenant` or `tenant_id` for better clarity.

Done.